### DISCUSSION TITLE
Lots of delete functions have no security dialog to prevent deletion

### DISCUSSION THREAD

**Discussion by tone2tone** [CONTENIDO/CONTENIDO#493] [General]
Quite often it happens to me that in Smarty template mode I click the wrong button and delete a template when I wanted nothing more than creating a new one. No dialog is shown for "do you really want to delete this file?" Same thing on other occasions.

I tried to work over the javascript routine for the smarty template deletion routine, but am not successful to add the aforementioned dialog, at least not contenido's own.

In class.html.link.php, from line 116 on, it reads:

```
    public function enableAutomaticParameterAppend() {
        return $this->setEvent('click', 'var doit = true; try { var i = get_registered_parameters() } catch (e) { doit = false; }; if (doit == true) { this.href += i; }');
    } 
```
I tried various ways to add Con.showConfirmation, but was only half way successful as in that the javascript dialog appeared and can be cancelled, but the "OK" button doesn't proceed with the functionality, so no deletion...

`		return $this->setEvent("click", "Con.showConfirmation('" . $confirmString . "',  function() {var doit = true; try { var i = get_registered_parameters() } catch (e) { doit = false; }; if (doit == true) { this.href += i; }});return false;");`

Don't know why the original code packed into the function() won't work. Not enough javascript experience here to fully get it.
Thoughts?

**Comment by muratpurc** [comment]
@tone2tone The CONTENIDO backend is far away from being a SPA (Single Page Application) or a PWA (Progressive Web App), but to match a bit with modern development, we should try to avoid the usage of inline scripting and inline css.

Ideally, HTML should contain the document structure, CSS the styles and JS the business logic.

There was a deeper change in the sources during the work on the ticket #35 , a lot of inline scripting has been moved from markup to the script section of the templates. This approach should be continued.

Later on, we can think about to port the script codes bit by bit into proper JavaScript modules/components.

**Comment by tone2tone** [comment]
@purcaholic My "problem" with what you say is that I do not really get it - no idea what the way is that I should proceed. I do not know the system enough to know whether the Con.showConfirmation thing is the way to go, or the get_registered-parameters thing, or something else. I would very much like to move behaviour into scripts, but on the other hand, I do not know where and how to find pre-configured functions and methods to use; in most cases it would require to completely change the way the buttons are set up. Sometimes existing routines rely on how text and buttons are coupled as sibling elements, sometimes I would need to know the id of the element (but don't)... and so on and so forth. Can you give a straight forward example (without being 100% correct) as YOU would resolve the problem with the smarty templates and producing a WORKING dialog box (visually inline with backend dialog) that works for OK and Cancel? Otherwise I am struggling. Need more practical direction here.

**Comment by gonatan** [comment]
@tone2tone Could you give a hint where the delete confirmation works as desired and where not?

I guess that several different solutions have been implemented in order to solve the same problem due to the vast amount of developers that have contributed to CONTENIDO.

**Comment by tone2tone** [comment]
@gonatan That was more or less the question I asked purcaholic as I have no idea which is the way to proceed. There are popups for confirmation e.g. when deleting a category or article or such, but not when you delete a smarty template. The javascript routines to make this buttons do what is needed are very different and it is not easy to change it, as some of the table based templates in the module section work with very different and distinguished routines that need a general addition, but then some dialogs appear even when creating a new template where you don't need them.
